libraries/libfc/include/fc/crypto/sha256.hpp

```diff
@@ -147,5 +147,12 @@ namespace boost
        }
     };
 }
+
+namespace fc {
+   inline size_t hash_value(const fc::sha256& s) {
+      return boost::hash<fc::sha256>()(s);
```

Comment:
I would have thought that the version taking one of the `uint64_t` (like `return v._hash[3];`) would be just as good as rehashing the four values.

Reply:
Isn't this just calling that version? It appears to be due to `template<> struct hash<fc::sha256>` not being `const`, whereas this one is `const`. afaict both const & non-const are needed. I suggest we clean it up in a separate PR

Reply:
Yes, sorry, I missed that we defined it for the boost namespace.